src_c/font.c

```diff
@@ -514,12 +514,6 @@ font_render(PyObject *self, PyObject *args)
             return RAISE(PyExc_ValueError,
                          "A null character was found in the text");
         }
-        if (utf_8_needs_UCS_4(astring)) {
```

Comment:
This PR is looking good! Having `pygame.font` being able to handle more unicode is a good thing. Now for review, rather than completely removing this block, it would be better if you put some handling for SDL_TTF < 2.0.15 (maybe keep displaying this error message in that case). Also, another minor code nitpick, `utf_8_needs_UCS_4` function defined in this file is not used anywhere now, and can be removed. This change needs docs.

Also, the `ftfont` module will become irrelevant after this change, so we have to see what to do about it

Reply:
Okay, I made a preprocessor condition which only "removes" the UCS-4 restriction if SDL_ttf is [above] 2.0.15

Reply:
Anyway, I will mark this convo as resolved. Since this change only take part on SDL_ttf >= 2.0.15, which pygame that uses SDL_ttf versions below 2.0.15 might still want to use `utf_8_needs_UCS_4` and `ftfont` for reverse compatibility sake (Which I guess is a common theme with illume keeping Python 2, kek). You can unresolve this convo if you have something to add.